### DISCUSSION TITLE
Invalidate disabled infinite query

### DISCUSSION THREAD

**Discussion by ranshine** [TanStack/query#3494] [General]
I am having some infinite query list and i am trying to delete one item from that list , after that i am calling invalidate query still it is not calling.

fetching list:

const { data, isLoading, isError, hasNextPage, fetchNextPage } =
  useInfiniteQuery("posts", fetchPosts, {
    refetchOnMount: false,
    refetchOnWindowFocus: false,
    enabled:false,
    getNextPageParam: (lastPage, pages) => {
      if (lastPage.nextPage < lastPage.totalPages) return lastPage.nextPage;
      return undefined;
    },
  });

delete item:

const mutation = useMutation(deleteItem, {
   onSuccess: () => {
     queryClient.invalidateQueries('posts')
   },
 })

not refreshing list

**Comment by TkDodo** [comment]
disabled queries can only refetch in two ways:
- by enabling them
- by calling `refetch` returned from `useInfiniteQuery`. 

That is per design. Why is your query disabled?

**Reply by ranshine** [reply]
if i enable useInfiniteQuery, if i try to refetch some dummy request query parameters calls are going

**Reply by TkDodo** [reply]
i don't quite follow. usually, you disable a query _until_ all parameters are ready...

**Reply by ranshine** [reply]
how can i use refetch in other components which is a parameter useInifiniteQuery

**Reply by ranshine** [reply]
i can use refetchqueries for disable query right like this await queryClient.refetchQueries(“posts”);

**Reply by ranshine** [reply]
if i call invalidate queries for useInfinitequery, some dummy calls are going, if i do resetqueries before invalidate query then it is working fine

**Reply by TkDodo** [reply]
sorry, I don't really understand what you are trying to achieve. You are asking very specific questions. It would be better if you'd explain, on a higher level, what problem you have that you are trying to solve...

I can see "refetch with parameter", and that usually boils down to:
- putting the dependencies into the queryKey
- setting enabled conditionally

**Comment by netojose** [comment]
I spent a lot of time and my infiniteQuery wasn't being invalidated and or refetched via queryClient. But, in the end, reading this, I could to understand if the query is *enabled: false*, it's impossible achieve this, except following the steps mentioned [above](https://github.com/TanStack/query/discussions/3494#discussioncomment-2542376).

Thanks for the explanation, @TkDodo .
